### DISCUSSION TITLE
ruby-3.1.2编译问题，

### DISCUSSION THREAD

**Discussion by bauw2008** [vernesong/OpenClash#2583] [General]
官方5.15，openclash v0.45.41-beta。

gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

./tool/file2lastrev.rb:6:in `require': cannot load such file -- optparse (LoadError)
	from ./tool/file2lastrev.rb:6:in `<main>'
uncommon.mk:1191: recipe for target 'revision.tmp' failed
make[4]: *** [revision.tmp] Error 1
make[4]: Leaving directory '/home/bauw/openwrt23/build_dir/hostpkg/ruby-3.1.2'
Makefile:1155: recipe for target '/home/bauw/openwrt23/build_dir/hostpkg/ruby-3.1.2/.built' failed
make[3]: *** [/home/bauw/openwrt23/build_dir/hostpkg/ruby-3.1.2/.built] Error 2
make[3]: Leaving directory '/home/bauw/openwrt23/feeds/packages/lang/ruby'
time: package/feeds/packages/ruby/host-compile#0.60#0.21#0.80
    ERROR: package/feeds/packages/ruby [host] failed to build.

./tool/file2lastrev.rb:6:in `require': cannot load such file -- optparse (LoadError)错误，有没有不拉源码修复的办法。

另外ruby-3.1.2没有ruby-dbm依赖，使用好像没什么问题。不知道这个有没有影响。

**Comment by vernesong** [comment]
源代码问题无法提供帮助，只要能正常使用就行

**Reply by bauw2008** [reply]
好的，感谢你的回复。
